### PR TITLE
[#170890360] Cab_ride_driver model and Validation 

### PR DESCRIPTION
**Pull request by Durgesh00**
Generated Cab,Driver and Ride model with all the validation and used array to store the values of routes.

### CODE REVIEW

File: app/models/cab.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class Cab < ApplicationRecord
+  belongs_to :driver
+  has_many :rides
+  validates :Vehicle_number, :cab_capacity, presence: true
```
Comment (smriti154) [inline app/models/cab.rb]: Change the case of vehicle_number

File: app/models/cab.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  validates :Vehicle_number, length: {in: 10..10}
```
Comment (smriti154) [inline app/models/cab.rb]: Change the case of vehicle_number

File: app/models/cab.rb (hunk 1 above)
Comment (Durgesh00) [reply app/models/cab.rb]: Done!

File: app/models/cab.rb (hunk 2 above)
Comment (Durgesh00) [reply app/models/cab.rb]: Done!!

File: db/migrate/20200127064346_create_cabs.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class CreateCabs < ActiveRecord::Migration[5.2]
+  def change
+    create_table :cabs do |t|
+      t.integer :cab_capacity   
+      t.string :Vehicle_number
```
Comment (smriti154) [inline db/migrate/20200127064346_create_cabs.rb]: Vehicle_number should not be in capital

File: app/models/driver.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+class Driver < ApplicationRecord
+  has_one :cab
+  validates :first_name, :last_name, :mobile_number, :licence_number, presence: true
+  validates :mobile_number, uniqueness: true, length: {in: 10..10}, numericality: {only_integer: true}
+  validates :licence_number, length: {in: 15..15}
+  before_save do
+    self.first_name = first_name.titleize
+    self.last_name = last_name.titleize
+  end
+end
```
Comment (smriti154) [inline app/models/driver.rb]: @Durgesh00 Discard the driver model from this PR. We will discuss, how the driver model will be.

File: app/presenters/v1/base_presenter.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
```
Comment (smriti154) [inline app/presenters/v1/base_presenter.rb]: This file can be removed

File: app/models/driver.rb (hunk 4 above)
Comment (Durgesh00) [reply app/models/driver.rb]: Done

File: app/presenters/v1/base_presenter.rb (hunk 5 above)
Comment (Durgesh00) [reply app/presenters/v1/base_presenter.rb]: done

File: db/migrate/20200127064346_create_cabs.rb (hunk 3 above)
Comment (Durgesh00) [reply db/migrate/20200127064346_create_cabs.rb]: done

**Pull request closed without merging**